packages/framework/ai-collab/src/test/explicit-strategy/agentEditingReducer.spec.ts

```diff
@@ -1412,3 +1431,732 @@ describe("applyAgentEdit", () => {
+					place: "after",
+				},
+			};
+			const result = applyAgentEdit(insertEdit, idGenerator, schema.definitions);
+
+			const expectedNewInsertIndex = 1;
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const newlyInsertedNode = view.root.rootVectors.at(expectedNewInsertIndex)!;
+
+			const expectedUDiff: InsertDiff = {
+				type: "insert",
+				nodePath: [
+					{
+						shortId: Tree.shortId(newlyInsertedNode),
+						schemaIdentifier: Tree.schema(newlyInsertedNode).identifier,
+						parentField: expectedNewInsertIndex,
+					},
+					{
+						shortId: undefined, // Tree.shortId() would return undefined for the array since it has no sf.identifier field
+						schemaIdentifier: Tree.schema(view.root.rootVectors).identifier,
+						parentField: "rootVectors",
+					},
+					{
+						shortId: Tree.shortId(view.root),
+						schemaIdentifier: Tree.schema(view.root).identifier,
+						parentField: "rootFieldKey",
+					},
+				],
+				aiExplanation: insertEdit.explanation,
+				nodeContent: JSON.parse(JSON.stringify(newlyInsertedNode)) as unknown,
+			};
+			assert.deepEqual(result.uiDiff, expectedUDiff);
+		});
+
+		it("insert non-primitive into array node via ArrayPlace", () => {
+			const { view, schema } = initializeTree();
+			const targetVectorArrayParentId = idGenerator.getId(view.root) ?? fail("ID expected.");
+
+			const insertEdit: Insert = {
+				explanation: "Insert a vector",
+				type: "insert",
+				content: { [typeField]: TestVector.identifier, x: 2, y: 3, z: 4 },
+				destination: {
+					type: "arrayPlace",
+					parentId: targetVectorArrayParentId,
+					field: "rootVectors",
+					location: "start",
+				},
+			};
+			const result = applyAgentEdit(insertEdit, idGenerator, schema.definitions);
+
+			const expectedNewInsertIndex = 0;
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const newlyInsertedNode = view.root.rootVectors.at(expectedNewInsertIndex)!;
+
+			const expectedUDiff: InsertDiff = {
+				type: "insert",
+				nodePath: [
+					{
+						shortId: Tree.shortId(newlyInsertedNode),
+						schemaIdentifier: Tree.schema(newlyInsertedNode).identifier,
+						parentField: expectedNewInsertIndex,
+					},
+					{
+						shortId: undefined, // Tree.shortId() would return undefined for the array since it has no sf.identifier field
+						schemaIdentifier: Tree.schema(view.root.rootVectors).identifier,
+						parentField: "rootVectors",
+					},
+					{
+						shortId: Tree.shortId(view.root),
+						schemaIdentifier: Tree.schema(view.root).identifier,
+						parentField: "rootFieldKey",
+					},
+				],
+				aiExplanation: insertEdit.explanation,
+				nodeContent: JSON.parse(JSON.stringify(newlyInsertedNode)) as unknown,
+			};
+			assert.deepEqual(result.uiDiff, expectedUDiff);
+		});
+		// it("insert primitive into array node via ObjectPlace - NOT SUPPORTED - see agentEditReducer line 143, typeof allowedType === function prevents this", () => {});
```

Comment:
Nit: I would probably replace these comments with a simple note that moving primitives isn't supported. The specific test syntax and references to line numbers will likely become out of date over time and may be confusing.

Reply:
I appreciate your thoroughness though!!